r/R/dplyr-funcs-datetime.R

```diff
@@ -326,7 +326,7 @@ register_bindings_datetime_conversion <- function() {
     # => we only cast when we want the behaviour of the base version or when
     # `tz` is set (i.e. not NULL)
     if (call_binding("is.POSIXct", x)) {
-      x <- build_expr("cast", x, options = cast_options(to_type = timestamp(timezone = tz)))
+      x <- build_expr("cast", x, options = cast_options(to_type = timestamp(x$type()$unit(), timezone = tz)))
```

Comment:
It seems that `"ns"` needs to be hard-coded as `x$tyep()$unit()` would not be exist if x was an R object.